apisix/core/config_etcd.lua

```diff
@@ -262,9 +262,15 @@ local function do_run_watch(premature)
             cancel_watch(http_cli)
             break
         end
-        if rev > watch_ctx.rev then
-            watch_ctx.rev = rev + 1
+
+        if rev < watch_ctx.rev then
+            log.error("received smaller revision, rev=", rev, ", watch_ctx.rev=",
+                      watch_ctx.rev,". etcd may be restarted. resyncing....")
+            produce_res(nil, "restarted")
+            cancel_watch(http_cli)
+            break
         end
+        watch_ctx.rev = rev + 1
```

Comment:
what will happen when `rev == watch_ctx.rev` ?

Reply:
updated the code.
1. Smaller revision -> Cause resync
2. Equal revision -> Do nothing like previous
3. Bigger revision -> Increment watch_ctx.rev